File: algorithm/acwing/LanQiaoBeiDayOne2023/src/Week1/day5/AcWing1208/main.cpp
#include <cstring>
#include <iostream>
#include <algorithm>

using namespace std;
/**
 * @Author 袁佳林
 * @Description
 * @Date 2023/2/25 10:55
 */
const int N = 110;

int n;
char start[N], aim[N];

void turn(int i)
{
    if (start[i] == '*') start[i] = 'o';
    else start[i] = '*';
}

int main()
{
    cin >> start >> aim;
    n = strlen(start);

    int res = 0;
    for (int i = 0; i < n - 1; i ++ )
        if (start[i] != aim[i])
        {
            turn(i), turn(i + 1);
            res ++ ;
        }

    cout << res << endl;
    return 0;
}

File: algorithm/acwing/LanQiaoBeiDayOne2023/src/Week1/day5/AcWing1208/main.py
#author:袁佳林
ori = list(input())
tar = list(input())
res = 0
for i in range(len(ori) - 1):
    if ori[i] != tar[i]:
        res += 1
        ori[i + 1] = '*' if ori[i + 1] == 'o' else 'o'
print(res)